clap.schema.json

```diff
@@ -0,0 +1,429 @@
+        "required_if": {
+          "$ref": "#/structures/tuple2StrStr"
+        },
+        "required_ifs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/structures/tuple2StrStr"
+          }
+        },
+        "required_unless": {
+          "type": "string"
+        },
+        "required_unless_all": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "required_unless_one": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "requires": {
+          "type": "string"
+        },
+        "requires_all": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "requires_if": {
+          "$ref": "#/structures/tuple2StrStr"
+        },
+        "requires_ifs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/structures/tuple2StrStr"
+          }
+        },
+        "setting": {
+          "type": "string"
+        },
+        "settings": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "short": {
+          "$ref": "#/definitions/shortArg"
+        },
+        "short_alias": {
+          "$ref": "#/definitions/shortArg"
+        },
+        "short_aliases": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/shortArg"
+          },
+          "uniqueItems": true
+        },
+        "takes_value": {
+          "type": "boolean"
+        },
+        "use_delimiter": {
+          "type": "boolean"
+        },
+        "value_name": {
+          "type": "string"
+        },
+        "value_names": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "value_terminator": {
+          "type": "string"
+        },
+        "value_delimiter": {
+          "type": "string"
+        },
+        "visible_alias": {
+          "type": "string"
+        },
+        "visible_aliases": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "visible_short_alias": {
+          "type": "string"
+        },
+        "visible_short_aliases": {
+          "$ref": "#/structures/arrayStrUnique"
+        }
+      }
+    },
+    "argGroup": {
+      "properties": {
+        "arg": {
+          "type": "string"
+        },
+        "args": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "conflicts_with": {
+          "type": "string"
+        },
+        "conflicts_with_all": {
+          "$ref": "#/structures/arrayStrUnique"
+        },
+        "multiple": {
+          "type": "boolean"
+        },
+        "required": {
+          "type": "boolean"
+        },
+        "requires": {
+          "type": "string"
+        },
+        "requires_all": {
+          "$ref": "#/structures/arrayStrUnique"
+        }
+      },
+      "type": "object"
+    },
+    "shortArg": {
+      "maxLength": 1,
+      "minLength": 1,
+      "pattern": "^[^-]$",
+      "type": "string"
+    }
+  },
+
+  "type": "object",
+  "TODO:": "set 'name' as required",
```

Comment:
What's this?

Reply:
Because if I set it here, it will be required for subcommands as well, which is incorrect